feat(docs): Section about UP4W components deployment and updates (#1310)
canonical/ubuntu-pro-for-wsl: docs/explanation/ref-arch-explanation.md

* **Go**: Windows agent and WSL Pro service
* **Flutter**: GUI front end
* **gRPC**: communication between back end and front end

### Deployment and updates

The WSL Pro service, a component that runs inside instances of Ubuntu on WSL, is
distributed as a Debian package from the Ubuntu archive. It receives
automatic updates and upgrades that are typical for Ubuntu packages.
The Pro service comes pre-installed on the most recent LTS releases of Ubuntu on WSL.

The Windows components of Ubuntu Pro for WSL are distributed in a single
[MSIX](https://learn.microsoft.com/en-us/windows/msix/overview) package.
The package is available through the Microsoft Store, our download page, and as GitHub
Release assets. Together with the MSIX package, we also publish an
[`AppInstaller` file](https://learn.microsoft.com/en-us/windows/msix/app-installer/app-installer-file-overview)
that Windows treats as an installable file. The file is later used to track and fetch
updates for the Ubuntu Pro for WSL app, if it has not been installed from the Microsoft
Store. This a common scenario in many corporate environments, where access to the Microsoft Store may be restricted.

> On Windows 11, packages installed by any of these methods will receive automatic updates, which is guaranteed by

Author: Also the lines in the were too long in the source format, I'm breaking at 80 columns to be consistent with the rest of the documentation source.

* **Go**: Windows agent and WSL Pro service
* **Flutter**: GUI front end
* **gRPC**: communication between back end and front end

### Deployment and updates

> The wsl-pro-service, component running inside the Ubuntu on WSL instances, is

Reviewer: ```suggestion
The WSL Pro service, a component that runs inside instances of Ubuntu on WSL, is
```

> distributed as a Debian package from the Ubuntu archive, thus it features
> automatic updates and upgrades as usual for Ubuntu packages in general.

Reviewer: ```suggestion
distributed as a Debian package from the Ubuntu archive. It receives
automatic updates and upgrades that are typical for Ubuntu packages.
```

> It comes pre-installed by default on the most recent LTS releases.

Reviewer: ```suggestion
The Pro service comes pre-installed on the most recent LTS releases of Ubuntu on WSL.
```

> The Windows components are distributed in a single

Reviewer: ```suggestion
The Windows components of Ubuntu Pro for WSL are distributed in a single
```

> [MSIX](https://learn.microsoft.com/en-us/windows/msix/overview) package
> made available via the Microsoft Store, our download page and as GitHub
> Release assets. Together with the MSIX package we also publish an

Reviewer: ```suggestion
[MSIX](https://learn.microsoft.com/en-us/windows/msix/overview) package.
The package is available through the Microsoft Store, our download page, and as GitHub
Release assets. Together with the MSIX package, we also publish an
```

Do we need links for the download page and release assets (repo)?

Author: I link they fit well here, but for now we don't have them yet as we didn't release after implementing this mechanism.

[`AppInstaller` file](https://learn.microsoft.com/en-us/windows/msix/app-installer/app-installer-file-overview)

> that Windows treats as an installable file and later uses to track and fetch
> updates for the Ubuntu Pro for WSL app when not installed from the Microsoft
> Store, a common scenario in many corporate environments.

Reviewer: ```suggestion
that Windows treats as an installable file. The file is later used to track and fetch
updates for the Ubuntu Pro for WSL app, if it has not been installed from the Microsoft
Store. This a common scenario in many corporate environments, where access to the Microsoft Store may be restricted.
```

> All installation methods feature automatic updates on Windows 11 guaranteed by
> the operating system. Windows 10 specifically has limited support for automatic
> updates of MSIX packages installed directly. Instead, users should prefer
> installing the Windows application either from Microsoft Store or from the
> `AppInstaller` file that accompanies the MSIX package to ensure automatic
> updates.

Reviewer: ```suggestion
On Windows 11, packages installed by any of these methods will receive automatic updates, which is guaranteed by
the operating system. In comparison, Windows 10 has limited support for automatic
updates of MSIX packages that have been installed directly. To ensure automatic updates on Windows 10, it is preferable to
install the Ubuntu Pro for WSL application through the Microsoft Store or using the
`AppInstaller` file that accompanies the MSIX package.
```

I think that this should be in its own paragraph to break up the text a bit.
